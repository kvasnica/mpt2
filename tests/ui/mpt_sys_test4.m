function mpt_sys_test4

% tests whether we give a nice error message when input is an MLD structure
% generated by mpt_pwa2mld():
%

opt_sincos
S = mpt_pwa2mld(sysStruct);
lasterr('');
try
    % this should not work
    s = mpt_sys(S);
    worked = 1;
catch
    worked = 0;
end
mbg_assertfalse(worked);
mbg_assertcontains(lasterr, 'Cannot generate PWA representation, see message above.');


% after disabling the PWA conversion it should work
s = mpt_sys(S, 'nopwa');
