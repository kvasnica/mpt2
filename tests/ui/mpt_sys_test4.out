This is not a valid HYSDEL-generated MLD model, cannot convert it to PWA form.
Use "mpt_sys(S, 'nopwa')" to disable the automatic PWA conversion.
